src/overlays/actors/ovl_Boss_01/z_boss_01.c

```diff
@@ -246,191 +647,2935 @@ ActorInit Boss_01_InitVars = {
+    &gOdolwaThurstAttackAnim,    // ODOLWA_WAIT_THRUST_ATTACK,
+    &gOdolwaDoubleSlashAnim,     // ODOLWA_WAIT_DOUBLE_SLASH,
+    &gOdolwaSideToSideHopAnim,   // ODOLWA_WAIT_SIDE_TO_SIDE_HOP,
+    &gOdolwaSideToSideDanceAnim, // ODOLWA_WAIT_SIDE_TO_SIDE_DANCE,
+    &gOdolwaSpinDanceAnim,       // ODOLWA_WAIT_SPIN_DANCE
+    &gOdolwaJumpDanceAnim,       // ODOLWA_WAIT_JUMP_DANCE
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_01/func_809D44C0.s")
+typedef struct {
+    /* 0x00 */ Vec3f pos;
+    /* 0x0C */ Vec3f velocity;
+    /* 0x18 */ f32 gravity;
+    /* 0x1C */ s16 rotX;
+    /* 0x1E */ s16 rotY;
+    /* 0x20 */ char unk_20[0x8];
+    /* 0x28 */ u8 type;
+    /* 0x2A */ s16 timer;
+    /* 0x2C */ s16 isFallingBlockFragment;
+    /* 0x2E */ s16 angularVelocityX;
+    /* 0x30 */ s16 angularVelocityY;
+    /* 0x34 */ f32 scale;
+    /* 0x38 */ f32 alpha;
+} OdolwaEffect; // size = 0x3C
+
+/**
+ * Keeps track of how many bugs are currently spawned and alive. When there are four or fewer bugs alive, Odolwa can
+ * summon more bugs or falling blocks.
+ */
+s16 sBugCount;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_01/func_809D4668.s")
+/**
+ * Pointer to the main Odolwa instance so that bugs and effects can check and manipulate his state.
+ */
+Boss01* sOdolwa;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_01/Boss01_Update.s")
+/**
+ * Pointer to the swarm of moths that Odolwa can summon.
+ */
+EnTanron1* sMothSwarm;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_01/func_809D519C.s")
+/**
+ * Odolwa draws his sword trail manually by manipulating these variables during his horizontal and vertical slashes.
+ * Note that the position of the sword is offset from Odolwa's position.
+ */
+f32 sSwordTrailPosX;
+f32 sSwordTrailPosY;
+f32 sSwordTrailPosZ;
+f32 sSwordTrailRotX;
+f32 sSwordTrailRotY;
+f32 sSwordTrailRotZ;
+f32 sSwordTrailAlpha;
+
+/**
+ * If the intro cutscene was skipped (because it was already watched), this timer will be used to determine when to
+ * start playing the boss theme.
+ */
+u8 sOdolwaMusicStartTimer;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_01/func_809D5584.s")
+/**
+ * When Odolwa is damaged or dazed, the sound effect will play from this position. It is set to his projectedPos.
+ */
+Vec3f sDamageSfxPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_01/func_809D55CC.s")
+s32 sOdolwaSeed1;
+s32 sOdolwaSeed2;
+s32 sOdolwaSeed3;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_01/Boss01_Draw.s")
+/**
+ * Stores all of Odolwa's effects. Note that the first entry in this array is always reserved for the ring of fire, so
+ * falling blocks can only be placed from the second entry onwards.
+ */
+OdolwaEffect sOdolwaEffects[ODOLWA_EFFECT_COUNT];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_01/func_809D5B0C.s")
+void Boss01_InitRand(s32 seedInit1, s32 seedInit2, s32 seedInit3) {
```

Comment:
In OoT, this is called `InitRand` in literally every actor *except* Fishing. Because Fishing was done first in MM, it looks like `SeedRand` managed to creep its way in here. I'd really rather just change those other two files than change this one; not only is `InitRand` far more common in OoT, but it also better describes what's going on here.

Reply:
Yeah, that's fine with me